backend/src/controllers/auth.controller.ts

```diff
@@ -0,0 +1,106 @@
+import { Request, Response } from "express";
+import { UserService } from "../services/user.service";
+import { RegisterUserDto } from "../dto/user/registerUser.dto";
+import { User } from "../models";
+
+export class AuthController {
+
+    private userService: UserService;
+
+    constructor() {
+        this.userService = new UserService;
+    }
+
+    public login = async (request: Request, response: Response) => {
+        try {
+            return response.status(200).json({
+                success: true,
+                message: 'Inicio de sesión exitoso!',
+                data: null,
+            });
+        } catch (error: any) {
+            return response.status(500).json({
+                success: false,
+                message: "Error al iniciar sesión!",
+                data: null,
+            });
+        }
+    }
+
+    public register = async (request: Request, response: Response) => {
+        try {
+            const body = request.body;
+            const user = await this.userService.registerUser(new RegisterUserDto(
+                body.first_name, body.last_name, body.email, body.password, body.role_id, body.phone ?? null, body.data
+            ));
```

Comment:
Esto se podría refactorizar algo así:
```suggestion
            const body = request.body as RegisterUserDto;
            const user = await this.userService.registerUser(body);
```
Hay que tener en cuenta que el DTO aquí no sirve del todo para validar, como en otros frameworks que lo acoplas a un pipe.
Para validaciones contamos con "express-validator".